server/doltgres_handler.go

```diff
@@ -0,0 +1,540 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"regexp"
+	"runtime/trace"
+	"sync"
+	"time"
+
+	sqle "github.com/dolthub/go-mysql-server"
+	"github.com/dolthub/go-mysql-server/server"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/dolthub/vitess/go/mysql"
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+	"github.com/jackc/pgx/v5/pgproto3"
+	"github.com/sirupsen/logrus"
+
+	"github.com/dolthub/doltgresql/postgres/messages"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// Result represents a query result.
+type Result struct {
+	Fields       []pgproto3.FieldDescription `json:"fields"`
+	Rows         []Row                       `json:"rows"`
+	RowsAffected uint64                      `json:"rows_affected"`
+}
+
+// Row represents a single row value in bytes format.
+// |val| represents array of a single row elements,
+// which each element value is in byte array format.
+type Row struct {
+	val [][]byte
+}
+
+const rowsBatch = 128
+
+// DoltgresHandler is a handler uses SQLe engine directly
+// running Doltgres specific queries.
+type DoltgresHandler struct {
+	e                 *sqle.Engine
+	sm                *server.SessionManager
+	readTimeout       time.Duration
+	encodeLoggedQuery bool
+}
+
+var _ = DoltgresHandler{}
```

Comment:
This isn't a type assertion :)

Needs to be like

```go
var _ InterfaceType = DoltgresHandler{}
```

The idea is that you're 1) enforcing at compile time that this type implements the interface, 2) documenting via code which interfaces it implements